spec/account/account-manager.spec.js

```diff
@@ -0,0 +1,154 @@
+                        mfa: accountInfo.mfa,
+                        salt: accountInfo.salt
+                    });
+                });
+                this.complete = jasmine.createSpy("complete");
+                this.complete.andCallFake(() => {
+                    return promiseMock.resolve(true);
+                });
+                this.get = jasmine.createSpy("get");
+                this.get.andCallFake(() => {
+                    return promiseMock.resolve({
+                        mfa: "339r93939303093"
+                    });
+                });
+            }
+        }
+        class HotpFake {
+            constructor() {
+                this.generateSecret = jasmine.createSpy("generateSecret");
+                this.generateSecret.andCallFake(() => {
+                    return promiseMock.resolve("thisisasecrectcodefrommfa");
+                });
+                // This is a sync call
+                this.verifyToken = jasmine.createSpy();
+                this.verifyToken.andCallFake((key, token, opts) => {
+                    if (token === "987654") {
+                        return false;
+                    }
+
+                    return true;
+                });
+            }
+        }
+        config = {
+            account: {
+                completeLifetime: {
+                    months: 6
+                },
+                initiateLifetime: {
+                    hours: 1
+                }
+            },
+            hotp: {
+                previous: {
+                    afterDrift: 1,
+                    beforeDrift: 1,
+                    drift: 2
+                }
+            }
+        };
+        accountManager = new AccountManager(new AccountServiceFake, config, new HotpFake, otDateMock, randomMock, promiseMock);
+    });
+    describe(".initiate()", () => {
+        it("gets an object back from service", (done) => {
+            accountManager.initiate({
+                email: "some.one@example.net"
+            }).then((result) => {
+                expect(result).toEqual({
+                    accountId: jasmine.any(Buffer),
+                    email: "some.one@example.net",
+                    mfa: "thisisasecrectcodefrommfa",
+                    salt: jasmine.any(Buffer)
+                });
+                expect(result.accountId.length).toBe(24);
+                expect(result.salt.length).toBe(128);
+
+            }).then(done, done);
+        });
+    });
+    describe(".complete()", () => {
+        var expectError, fail;
+
+        beforeEach(() => {
+            expectError = (done, contains) => {
+                // Generate a function that asserts the result is an error
+                // and contains some text in the message.
+                return (err) => {
+                    expect(err).toEqual(jasmine.any(Error));
+                    expect(err.toString()).toContain(contains);
+                    done();
```

Comment:
1) This should just be a function, perhaps at line 91.
2) This is the third time we do this stuff.  It's time to make custom matchers and functions.  That should be a new task - don't hold up this patch but it should get addressed next.